llvm/docs/InstCombineContributorGuide.md

```diff
@@ -0,0 +1,467 @@
+# InstCombine contributor guide
+
+This guide lays out a series of rules that contributions to InstCombine should
+follow. **Following these rules will results in much faster PR approvals.**
+
+## Tests
+
+### Precommit tests
+
+Tests for new optimizations or miscompilation fixes should be pre-committed.
+This means that you first commit the test with CHECK lines prior to your fix.
```

Comment:
'prior to your fix' sounds a bit unclear imo. How about 'without your implementation'?